runtime/src/main/java/org/corfudb/runtime/collections/ImmutableCorfuTable.java

```diff
@@ -199,25 +203,29 @@ public Stream<java.util.Map.Entry<K, V>> entryStream() {
      * @return An Iterable of map entries satisfying this index query.
      */
     public <I> Iterable<java.util.Map.Entry<K, V>> getByIndex(@Nonnull final Index.Name indexName, I indexKey) {
-        return secondaryIndexesWrapper.contains(indexName.get(), indexKey)
-                .<Iterable<java.util.Map.Entry<K, V>>>map(TupleIterableWrapper::new)
-                .getOrElse(Collections.emptySet());
```

Comment:
The problem with TupleIterableWrapper::new here was just the mixing of the Iterator types.